marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-view.tsx

```diff
@@ -22,16 +22,50 @@ type Props = {
 const GraphView: React.FC<Props> = (props) => {
 
   const {modelingOptions, setSelectedEntity, closeSidePanelInGraphView} = useContext(ModelingContext);
+  const [nodeSuggested, setNodeSuggested] = useState(["a"]);
+  const [entitySelected, setEntitySelected] = useState("");
+  const [isEntitySelected, setIsEntitySelected] = useState(false);
 
   const publishIconStyle: CSSProperties = {
     width: "20px",
     height: "20px",
     fill: "currentColor",
   };
 
+  const handleFocus = () => {
+    setNodeSuggested([]);
+  };
+
+  const handleTypeaheadChange = (value: any) => {
+    setEntitySelected(value);
+    setIsEntitySelected(false);
+    if (value.length > 2) {
+      Object.keys(props.entityTypes).map((key) => {
+        let obj=nodeSuggested;
+        if (value && !nodeSuggested.includes(props.entityTypes[key]["entityName"]) && props.entityTypes[key]["entityName"].toLowerCase().indexOf(value.toLowerCase()) >= 0) {
+          obj.push(props.entityTypes[key]["entityName"]);
+        }
+        setNodeSuggested(obj);
```

Comment:
Is nodeSuggested the items that appear in the filter dropdown menu? If so, can we call them something that describes that, like "filterMenuSuggestions"?

Reply:
Replaced nodeSuggested  with filterMenuSuggestions